File: Classes/EVEOnlineAPI/Character/EVECharStandings.m
//
//  EVECharStandings.m
//  EVEOnlineAPI
//
//  Created by Artem Shimanski on 7/24/10.
//  Copyright 2010 __MyCompanyName__. All rights reserved.
//

#import "EVECharStandings.h"

@implementation EVEStandingsItem

+ (NSDictionary*) scheme {
	static NSDictionary* scheme = nil;
	if (!scheme)
		scheme = @{@"fromID":@{@"type":@(EVEXMLSchemePropertyTypeScalar)},
				   @"fromName":@{@"type":@(EVEXMLSchemePropertyTypeString)},
				   @"standing":@{@"type":@(EVEXMLSchemePropertyTypeDate)}};
	
	return scheme;
}

@end


@implementation EVEStandingsNPCStandings

+ (NSDictionary*) scheme {
	static NSDictionary* scheme = nil;
	if (!scheme)
		scheme = @{@"agents":@{@"type":@(EVEXMLSchemePropertyTypeRowset), @"class":[EVEStandingsItem class]},
				   @"npcCorporations":@{@"type":@(EVEXMLSchemePropertyTypeRowset), @"class":[EVEStandingsItem class], @"elementName": @"NPCCorporations"},
				   @"factions":@{@"type":@(EVEXMLSchemePropertyTypeRowset), @"class":[EVEStandingsItem class]}};
	
	return scheme;
}

@end

@implementation EVECharStandings

+ (NSDictionary*) scheme {
	static NSDictionary* scheme = nil;
	if (!scheme)
		scheme = @{@"characterNPCStandings":@{@"type":@(EVEXMLSchemePropertyTypeObject), @"class":[EVEStandingsNPCStandings class]}};
	
	return scheme;
}

@end

File: EVEAPI/EVEAPI/EVECharStandings.swift
//
//  EVECharStandings.swift
//  EVEAPI
//
//  Created by Artem Shimanski on 29.11.16.
//  Copyright © 2016 Artem Shimanski. All rights reserved.
//

import UIKit

public class EVEStandingsItem: EVEObject {
	public var fromID: Int64 = 0
	public var fromName: String = ""
	public var standing: Double = 0
	
	public required init?(dictionary:[String:Any]) {
		super.init(dictionary: dictionary)
	}
	
	public required init?(coder aDecoder: NSCoder) {
		super.init(coder: aDecoder)
	}
	
	override public func scheme() -> [String:EVESchemeElementType] {
		return [
			"fromID":EVESchemeElementType.Int64(elementName:nil, transformer:nil),
			"fromName":EVESchemeElementType.String(elementName:nil, transformer:nil),
			"standing":EVESchemeElementType.Double(elementName:nil, transformer:nil),
		]
	}
}

public class EVEStandingsNPC: EVEObject {
	public var agents: [EVEStandingsItem] = []
	public var npcCorporations: [EVEStandingsItem] = []
	public var factions: [EVEStandingsItem] = []
	
	public override init() {
		super.init()
	}
	
	public required init?(dictionary:[String:Any]) {
		super.init(dictionary: dictionary)
	}
	
	public required init?(coder aDecoder: NSCoder) {
		super.init(coder: aDecoder)
	}
	
	override public func scheme() -> [String:EVESchemeElementType] {
		return [
			"agents":EVESchemeElementType.Rowset(elementName: nil, type: EVEStandingsItem.self, transformer: nil),
			"npcCorporations":EVESchemeElementType.Rowset(elementName: "NPCCorporations", type: EVEStandingsItem.self, transformer: nil),
			"factions":EVESchemeElementType.Rowset(elementName: nil, type: EVEStandingsItem.self, transformer: nil),
		]
	}
}

public class EVECharStandings: EVEResult {
	public var npcStandings: EVEStandingsNPC = EVEStandingsNPC()
	
	public required init?(dictionary:[String:Any]) {
		super.init(dictionary: dictionary)
	}
	
	public required init?(coder aDecoder: NSCoder) {
		super.init(coder: aDecoder)
	}
	
	override public func scheme() -> [String:EVESchemeElementType] {
		return [
			"npcStandings":EVESchemeElementType.Object(elementName: "characterNPCStandings", type: EVEStandingsNPC.self, transformer: nil),
		]
	}
}

